src/dataset.rs

```diff
@@ -315,6 +317,39 @@ impl Dataset {
         }
     }
 
+    /// Builds overviews for the current `Dataset`. See [`GDALBuildOverviews`].
+    ///
+    /// # Arguments
+    /// * `resampling` - resampling method, as accepted by GDAL, e.g. `"CUBIC"`
+    /// * `overviews` - list of overview decimation factors, e.g. `&[2, 4, 8, 16, 32]`
+    /// * `bands` - list of bands to build the overviews for, or empty for all bands
+    ///
+    /// [`GDALBuildOverviews`]: https://gdal.org/doxygen/gdal_8h.html#a767f4456a6249594ee18ea53f68b7e80
+    pub fn build_overviews(
+        &self,
```

Comment:
In line with https://github.com/georust/gdal/pull/161, I wonder if @rmanoka or @jdroenner have thoughts about whether this should be &mut.

Reply:
i guess &mut is the better choice.